BangazonAPI/Controllers/ProductController.cs

```diff
@@ -0,0 +1,243 @@
+                                        Product.CustomerId AS 'Customer Id',
+                                        Product.ProductTypeId AS 'Product Type Id'
+                                        FROM Product 
+                                        LEFT JOIN Customer on Product.CustomerId = Customer.Id
+                                        LEFT JOIN ProductType on Product.ProductTypeId = ProductType.Id
+                                        WHERE Product.Id = @id";
+                    
+                    
+                    cmd.Parameters.Add(new SqlParameter("@id", id));
+                    SqlDataReader reader = cmd.ExecuteReader();
+                    Product product = null;
+                    if (reader.Read())
+                    {
+                        product = new Product
+                        {
+                            Id = reader.GetInt32(reader.GetOrdinal("Id")),
+                            ProductTypeId = reader.GetInt32(reader.GetOrdinal("Product Type Id")),
+                            CustomerId = reader.GetInt32(reader.GetOrdinal("Customer Id")),
+                            Price = reader.GetDecimal(reader.GetOrdinal("Price")),
+                            Title = reader.GetString(reader.GetOrdinal("Title")),
+                            Description = reader.GetString(reader.GetOrdinal("Description")),
+                            Quantity = reader.GetInt32(reader.GetOrdinal("Quantity")),
+                            Archived = reader.GetBoolean(reader.GetOrdinal("Archived"))
+                        };
+                    }
+                    reader.Close();
+                    return Ok(product);
+                }
+            }
+        }
+        [HttpPost]
+        public async Task<IActionResult> Post([FromBody] Product product)
+        {
+            using (SqlConnection conn = Connection)
+            {
+                conn.Open();
+                using (SqlCommand cmd = conn.CreateCommand())
+                {
+                    cmd.CommandText = @"INSERT INTO Product (ProductTypeId, CustomerId, Price, Title, Description, Quantity, Archived)
+                                        OUTPUT INSERTED.Id
+                                        VALUES (@productTypeId, @customerId, @price, @title, @description, @quantity, @archived)";
+                    cmd.Parameters.Add(new SqlParameter("@productTypeId", product.ProductTypeId));
+                    cmd.Parameters.Add(new SqlParameter("@customerId", product.CustomerId));
+                    cmd.Parameters.Add(new SqlParameter("@price", product.Price));
+                    cmd.Parameters.Add(new SqlParameter("@title", product.Title));
+                    cmd.Parameters.Add(new SqlParameter("@description", product.Description));
+                    cmd.Parameters.Add(new SqlParameter("@quantity", product.Quantity));
+                    cmd.Parameters.Add(new SqlParameter("@archived", product.Archived));
```

Comment:
This is a design decision you should discuss with your group-- do y'all want the end user to be able to create an archived product? Or should `archived` automatically be set to `false` when you create a product, and only changed to `true` if they run a soft delete? (How much do you trust your end user?) I think an argument could be made either way and the client doesn't have a preference about this as long as you can back up your reasoning.